### PR TITLE
Feature/partnership

### PR DESCRIPTION
**Pull request by chadgates**
Fix of previously failed PR https://github.com/abhishek-ram/pyas2-lib/pull/43)

Add callback function for partnerships, that takes precedence over individual org and partner lookup. This will be used to tackle changes of organizational certificates, when there can be multiple combinations of certificates for organization and partner at a given time: https://github.com/abhishek-ram/django-pyas2/issues/21.

Additional remark: that "swap" of certificates - I now favor a concept of "Roll-Over Certificates". Nevertheless, same requirement needed in pyas2-lib.

### CODE REVIEW

File: pyas2lib/as2.py [hunk 1]
```diff
@@ -592,6 +605,18 @@ def parse(self, raw_content, find_org_cb, find_partner_cb, find_message_cb=None)
             the partner did not request it.
         """
 
+        # Validate passed arguments
+        if not any(
```
Comment (abhishek-ram) [inline pyas2lib/as2.py:609]: add a test for this

File: pyas2lib/as2.py [hunk 2]
```diff
@@ -605,14 +630,17 @@ def parse(self, raw_content, find_org_cb, find_partner_cb, find_message_cb=None)
...
-            self.receiver = find_org_cb(org_id)
-            if not self.receiver:
-                raise PartnerNotFound(f"Unknown AS2 organization with id {org_id}")
-
             partner_id = unquote_as2name(as2_headers["as2-from"])
-            self.sender = find_partner_cb(partner_id)
-            if not self.sender:
-                raise PartnerNotFound(f"Unknown AS2 partner with id {partner_id}")
+            if find_org_partner_cb:
+                self.receiver, self.sender = find_org_partner_cb(org_id, partner_id)
```
Comment (abhishek-ram) [inline pyas2lib/as2.py:635]: do we not need raise PartnerNotFound here as well?

File: pyas2lib/as2.py (hunk 1 above)
Comment (chadgates) [reply pyas2lib/as2.py:609]: Done

File: pyas2lib/as2.py (hunk 2 above)
Comment (chadgates) [reply pyas2lib/as2.py:635]: Move the checks down and outside the if statement.

**Review by abhishek-ram — APPROVED** [review]

**Pull request merged**